ibazel/fsnotify_watcher/fsnotify_watcher.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
```

Comment:
Big lift ask, is this testable at all?

Reply:
Yes, I just added a test to ensure the proper calls to Add() and Remove() are performed when various UpdateAll commands are invoked. Let me know if there is other functionality you would like tests added for.